public/app/plugins/panel/trend/TrendPanel.tsx

```diff
@@ -0,0 +1,123 @@
+import React, { useMemo } from 'react';
+
+import { FieldType, PanelProps } from '@grafana/data';
+import { config, PanelDataErrorView } from '@grafana/runtime';
+import { KeyboardPlugin, TimeSeries, TooltipDisplayMode, TooltipPlugin, usePanelContext } from '@grafana/ui';
+import { findField } from 'app/features/dimensions';
+
+import { ContextMenuPlugin } from '../timeseries/plugins/ContextMenuPlugin';
+import { prepareGraphableFields, regenerateLinksSupplier } from '../timeseries/utils';
+
+import { PanelOptions } from './panelcfg.gen';
+
+export const TrendPanel = ({
+  data,
+  timeRange,
+  timeZone,
+  width,
+  height,
+  options,
+  fieldConfig,
+  replaceVariables,
+  id,
+}: PanelProps<PanelOptions>) => {
+  const { sync } = usePanelContext();
+
+  const info = useMemo(() => {
+    if (data.series.length > 1) {
+      return {
+        warning: 'Only one frame is supported, consider adding a join transformation',
+        frames: data.series,
+      };
+    }
+
+    let xFieldName = options.xField;
+
+    if (xFieldName) {
+      const f = findField(data.series[0], xFieldName);
+
+      if (!f) {
+        return {
+          warning: 'Unable to find field: ' + xFieldName,
+          frames: data.series,
+        };
+      }
+    } else {
+      // first number field
+      const f = data.series[0].fields.find((f) => f.type === FieldType.number);
+
+      if (!f) {
+        return {
+          warning: 'No numeric fields found for X axis',
+          frames: data.series,
+        };
+      }
+
+      xFieldName = f.name;
+    }
+
+    const frames = prepareGraphableFields(data.series, xFieldName, config.theme2);
+
+    return { frames, xFieldName };
```

Comment:
^^^ rather than needing to find it again, lets just set it as the first field here